#9 Contribution Documentation (#10)
GoogleChromeLabs/wp-sitemaps: docs/CONTRIBUTING.md

# Contributing

> Thank you for thinking about contributing to the WordPress sitemaps project! Contributions can either be made in the form of code or through input on tickets.

Reviewer: ```suggestion
Thank you for thinking about contributing to the Core Sitemaps project! Contributions can either be made in the form of code or through input on tickets.
```

We appreciate all contributions as they help to move the project forward.

## Contributions to Issues

> Tickets will often need input from multiple points of view. We invite you to share your POV by adding a comment to the relevent ticket.

Reviewer 2: spelling: "relevant"

```suggestion
Tickets will often need input from multiple points of view. We invite you to share your POV by adding a comment to the relevant ticket.
```

> If a ticket does not yet exist, you can create a new ticket and add it to the back log, but please do read: https://make.wordpress.org/core/2019/06/12/xml-sitemaps-feature-project-proposal/ and double check that what you are proposing is inline with what we are trying to achieve within this project's scope.

Reviewer 2: compound and non-compound words:

```suggestion
If a ticket does not yet exist, you can create a new ticket and add it to the backlog, but please do read: https://make.wordpress.org/core/2019/06/12/xml-sitemaps-feature-project-proposal/ and double check that what you are proposing is in line with what we are trying to achieve within this project's scope.
```

### Creating an Issue

- In Github, under 'Issues' click the green 'new issue' button.
- Give the issue a short but meaningful title.
- Give us much detail about the issue in the description box
- Give the issue a label. Either:
	- Type: Bug (Something isn't working)
	- Type: Enhancement (Enhancement to an existing feature)
	- Type: Feature (New feature)
	- Type: Spike (Task to better understand a requirement)
	- Type: Task (Specific task that may not be part of a given feature)

> - Assign the issue to the `Execution` project

Reviewer 3: We might want to omit project documentation from the public instructions and make this something that we handle internally. Curious what @mcshanea thinks.

Reviewer 4: Yeah, this can be left out of the public instructions. I would say external submissions all fall under a general "issue" and don't fall under any of the three categories here, nor do they need to worry about the project workflow, we take care of that in refinement.

Please don't assign the t-shirt labels to issues or the issues to milestones as those are for internal use to manage workflow.

## Developer Contributions

When contributing through code, each feature should be developed in a seperate branch.

> - Create a new branch, forked from `develop`.

Reviewer 3: Branching strategy might need to be updated based on #12, but this is ok for now.

issues if one exists.

When submitting a PR there are some items you should take note of:
- Does the code follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
- Did you include unit tests (if applicable)?
- Was your local copy recently pulled from `develop`, so it's a clean patch?

The title of the pull request should be descriptive, summarize what your request is about in 5-10 words at most.
For the main body of the pull request, describe what you have done, and what problem this solves.
Include screenshots if possible to help visualize what changes have been introduced, and reference any open
issues if one exists.

## Contribute to the Documentation

All of the documentation for this plugin lives at [/docs/](/docs/README.md)

To contribute to the documentation either follow the Contributions to Issues workflow above and leave a comment about your suggested change. Or, follow the Developer Contributions workflow, and create a pull request with your suggested changes in.

## Reporting Security Issues

> [TODO: Add this]

Reviewer: good poin, needs discussion possibly with Google.

Reviewer 3: We should tell people to report it to plugins@wordpress.org, following the advice of https://developer.wordpress.org/plugins/wordpress-org/plugin-security/reporting-plugin-security-issues/.

Probably just copy/paste the first two paragraphs from that page.
